### ISSUE TITLE
Other sites are using the kubernetes.io analytics site ID, try to prevent that?

### ISSUE THREAD

**Issue by chalin** [kubernetes/website#36011]
@jimangel @a-mccarthy @sftim @jeefy @nate-double-u - idea why these two GH-pages hosted sites are using the main kubernetes.io UA site ID?

- https://jamesdefabia.github.io
- https://pwittrock.github.io

Are these "legit" and expected uses?

As you can see from the data given below, these two sites got thousands of hits in the past 30 days:

> <img width="599" alt="image" src="https://user-images.githubusercontent.com/4140793/184869827-4346951f-0b63-4079-b958-98d3e940c1c9.png">

Would you like measures taken to make it less likely that folks who clone this repo and deploy to a production server will automatically use the kubernetes.io analytics site ID?

Related:

- #35299

**Comment by chalin** [comment]
> I think this is a good idea. I'm not sure who we deliver the GA code currently, but we should make it so other folks can't use it accidentally on their site.

👍 Ok, I have a solution, but would need access to the Netlify account. Can you help me with that @a-mccarthy? (I'll ping some folks on Slack about it too.)

**Comment by jimangel** [comment]
Hopefully additional context: https://github.com/kubernetes/website/pull/22998#issuecomment-673124327

@chalin we can provide Netlify access, but would you mind sharing the exact solution / "how we got here" for awareness?

**Comment by sftim** [comment]
> Would you like measures taken to make it less likely that folks who clone this repo and deploy to a production server will automatically use the kubernetes.io analytics site ID?

It sounds useful to make it a bit more difficult, but I'd only give this backlog priority.

**Comment by sftim** [comment]
If there's a simple thing and a more secure thing, let's do the simple thing. We're not aiming to protect against deliberate misuse (also, such misuse easily shows up in the stats, and is easy to filter out).

**Comment by chalin** [comment]
> If there's a simple thing and a more secure thing, let's do the simple thing. We're not aiming to protect against deliberate misuse

I agree. That has been what I was thinking as well.

> Hopefully additional context: https://github.com/kubernetes/website/pull/22998#issuecomment-673124327.

Thanks for the additional context @jimangel.

> @chalin we can provide Netlify access, but would you mind sharing the exact solution

The solution consists of the combo of #36010 and setting the real GA ID via `HUGO_SERVICES_GOOGLEANALYTICS_ID` as a Netlify environment variable.

This solution also prevents local deploys from emitting data to GA, unbeknownst to the developer.

**Comment by jimangel** [comment]
Awesome! Thanks for the details @chalin! We might be able to set that build variable in our netlify config without having to grant access (ex: https://github.com/kubernetes/website/blob/main/netlify.toml#L14).

Let us know if you think that would suffice!

**Comment by chalin** [comment]
@jimangel - I think that it would be better to set the ID via the Netlify account, and here's why: the scenario we'd like to avoid is for a developer / group to _inadvertently_ use the Kubernetes GA ID. If we add the ID to the `netlify.toml` file, then a user who simply clones this repo and publishes via Netlify will have their site send GA events to the Kubernetes GA site tag. We won't have that problem if we set the GA ID via the Kubernetes Netlify account.

**Comment by chalin** [comment]
@jimangel et all: In case it can expedite things, I don't _need_ access to the Netlify account. If someone who already has access is willing to add the require environment variable (immediately after #36010 is merged), then that is all we need.

**Comment by chalin** [comment]
FYI, I've implemented the approach being suggested here for OTel:

- https://github.com/open-telemetry/opentelemetry.io/issues/1626
- https://github.com/open-telemetry/opentelemetry.io/pull/1628

**Comment by chalin** [comment]
@jimangel @sftim - any update on this? I'd like to keep the ball rolling and, more importantly, to be clear on who has the ball now. So, ... as a next step should I be trying to find someone with access to the Netlify account to do the config for me, or should I be submitting an official request to gain access (if the latter, how do I do so -- and should I be asking these questions on #sig-docs-maintainers)?

**Comment by chalin** [comment]
Btw, I'm assuming that we _do_ want analytics from the older versions of the docs:

> <img width="260" alt="image" src="https://user-images.githubusercontent.com/4140793/186509505-31b37d6b-24eb-4203-8792-88f1e63a3b4f.png">

I'm not sure how these are setup. Are they different netlify accounts?

**Comment by sftim** [comment]
One Netlify account, multiple sites.

**Comment by sftim** [comment]
I would ask on Slack @chalin - have you already done that?

**Comment by chalin** [comment]
> I would ask on Slack @chalin - have you already done that? The channel for that is #sig-docs-maintainers.

Yes, thanks, I've asked. As I understand it some folks are on vacation, so it may take a while still.

**Comment by chalin** [comment]
The measures taken so far have been, for the main site, only add the GA4 ID via Netlify config. This way folks cloning this repo cannot inadvertently start using the GA4 ID. I've taken no such special measures for branch releases: though it could be implemented in a similar way as the main site, that would require some effort to setup (cf., e.g., #36334) and it seems less likely that folks will be cloning branch releases. So, I'm closing this issue as it seems sufficiently resolved.
